catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventToXmlElementConverter.java

```diff
@@ -213,8 +219,21 @@ public SaxEventToXmlElementConverter reset() {
      * @param uri    the namespace uri that is passed in by {@link SaxEventHandler}
      */
     public void addNamespace(String prefix, String uri) throws XMLStreamException {
-        out.setPrefix(prefix, uri);
         namespaces.put(prefix, uri);
+        namespaceStack.push(new ImmutablePair<>(prefix, uri));
+    }
+
+    public void removeNamespace(String prefix) {
+        Iterator<ImmutablePair<String, String>> iter = namespaceStack.iterator();
+        while (iter.hasNext()) {
+            Pair pair = iter.next();
+            if (pair.getKey()
+                    .equals(prefix)) {
+                iter.remove();
+                break;
```

Comment:
I don't know if this could ever happen, but what if there are multiple pairs with the same prefix?